packages/flutter_tools/test/project_file_invalidator_test.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/run_hot.dart';
+
+import 'src/common.dart';
+import 'src/context.dart';
+
+void main() {
+  group(ProjectFileInvalidator, () {
```

Comment:
Do any of these pretend to be Windows?

Reply:
Added a group of windows unit tests, caught a bug, and simplified the Uri/file handling